great_expectations/data_context/store/store.py

```diff
@@ -248,9 +248,12 @@ def add(self, key: DataContextKey, value: Any, **kwargs) -> None:
         """
         return self._add(key=key, value=value, **kwargs)
 
-    def _add(self, key: DataContextKey, value: Any, **kwargs) -> None:
+    def _add(self, key: DataContextKey, value: Any, **kwargs) -> Any:
         self._validate_key(key)
-        return self._store_backend.add(self.key_to_tuple(key), self.serialize(value), **kwargs)
+        output = self._store_backend.add(self.key_to_tuple(key), self.serialize(value), **kwargs)
+        if hasattr(value, "id") and hasattr(output, "id"):
+            value.id = output.id
+        return output
```

Comment:
Can we check to see if any child classes override this method? Would be good to use the single base class impl

Reply:
> Can we check to see if any child classes override this method? Would be good to use the single base class impl

Good call! we do override it in a few places, but they all call into this one, e.g. `super()._add(whatevs)`